src/store.ts

```diff
@@ -0,0 +1,52 @@
+interface iState {
```

Comment:
This store looks great. Can you add unit tests for this whole implementation

Reply:
Done!